test/minikube/minikube_base_database_test.go

```diff
@@ -369,11 +380,102 @@ func TestKubernetesBackupDatabase(t *testing.T) {
-		backupDatabase(t, namespaceName, admin0, "demo", &databaseOptions)
+		backupDatabase(t, namespaceName, admin0, "demo", false, &databaseOptions)
+	})
+
+	t.Run("nobackupSMBackup", func(t *testing.T) {
+		t.Skip("See: https://github.com/nuodb/nuodb-helm-charts/issues/64")
+		defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+		databaseOptions := helm.Options{
+			SetValues: map[string]string{
+				"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+				"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+				"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+				"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+				"backup.persistence.enabled":            "true",
+				"backup.persistence.size":               "1Gi",
+				"database.sm.HotCopy.replicas":          "0",
+				"database.sm.HotCopy.enabled":           "true",
+				"database.sm.noHotCopy.replicas":        "1",
+			},
+		}
+
+		databaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &databaseOptions)
+		opts := k8s.NewKubectlOptions("", "")
+		opts.Namespace = namespaceName
+		opt := testlib.GetExtractedOptions(&databaseOptions)
+		smPodName := fmt.Sprintf("sm-%s-nuodb-%s-%s-hotcopy-0", databaseChartName, opt.ClusterName, opt.DbName)
+		fmt.Printf(k8s.RunKubectlAndGetOutputE(t, opts, "label", "--overwrite", "pods", "-n", namespaceName, smPodName, "role=nohotcopy"))
+		populateCreateDBData(t, namespaceName, admin0)
+		defer testlib.Teardown(testlib.TEARDOWN_BACKUP)
+		backupDatabase(t, namespaceName, admin0, "demo", false, &databaseOptions)
 	})
+
+
+	t.Run("backupNoSMs", func(t *testing.T) {
+		defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+		databaseOptions := helm.Options{
+			SetValues: map[string]string{
+				"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+				"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+				"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+				"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+				"backup.persistence.enabled":            "true",
+				"backup.persistence.size":               "1Gi",
+			},
+		}
+
+		databaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &databaseOptions)
+
+		populateCreateDBData(t, namespaceName, admin0)
+
+		opts := k8s.NewKubectlOptions("", "")
+		opts.Namespace = namespaceName
+		databaseOptions.KubectlOptions = opts
+		opt := testlib.GetExtractedOptions(&databaseOptions)
+		ss := fmt.Sprintf("sm-%s-nuodb-%s-%s-hotcopy", databaseChartName, opt.ClusterName, opt.DbName)
+
+		fmt.Printf(k8s.RunKubectlAndGetOutputE(t, opts, "scale", "-n", namespaceName, "statefulset", ss, "--replicas=0"))
+		defer testlib.Teardown(testlib.TEARDOWN_BACKUP)
+		backupDatabase(t, namespaceName, admin0, "demo", true, &databaseOptions)
+	})
+
+
+	t.Run("killSMProcessTest", func(t *testing.T) {
```

Comment:
what does this test do?

Reply:
It kills the only backup SM while the initial backup job is in some indeterminate state (probably not started), pauses to let the SM restart and backup to complete. The backup should still verify correctly. Might be able to remove the sleep (the await timeout in backupdatabase might be enough.. will experiment with that).
It is not a super useful test in this current state. It would be better to wait for the initial backup job to enter running state, kill it then, and examine the results. But I have no idea how to validate this kind of situation from an automation point of view, it would need to be manually tested first.

Reply:
Looking at the logs for this test, it runs for 2 min and does not verify anything. Maybe we can improve on it in the future

Reply:
Agreed. At least it runs for 1 min now.